lighthouse-core/test/fraggle-rock/scenarios/api-test-pptr.js

```diff
@@ -141,7 +141,41 @@ describe('Fraggle Rock API', () => {
 
     it('should compute both snapshot & timespan results', async () => {
       const {page, serverBaseUrl} = state;
-      const result = await lighthouse.navigation({page, url: `${serverBaseUrl}/index.html`});
+      const result = await lighthouse.navigation({page, requestor: `${serverBaseUrl}/index.html`});
+      if (!result) throw new Error('Lighthouse failed to produce a result');
+
+      const {lhr} = result;
+      const {auditResults, failedAudits, erroredAudits} = getAuditsBreakdown(lhr);
+      // TODO(FR-COMPAT): This assertion can be removed when full compatibility is reached.
+      expect(auditResults.length).toMatchInlineSnapshot(`152`);
+      expect(erroredAudits).toHaveLength(0);
+
+      const failedAuditIds = failedAudits.map(audit => audit.id);
+      expect(failedAuditIds).toContain('label');
+      expect(failedAuditIds).toContain('errors-in-console');
+
+      // Check that network request information was computed.
+      expect(lhr.audits).toHaveProperty('total-byte-weight');
+      const details = lhr.audits['total-byte-weight'].details;
+      if (!details || details.type !== 'table') throw new Error('Unexpected byte weight details');
+      expect(details.items).toMatchObject([{url: `${serverBaseUrl}/index.html`}]);
+
+      // Check that performance metrics were computed.
+      expect(lhr.audits).toHaveProperty('first-contentful-paint');
+      expect(Number.isFinite(lhr.audits['first-contentful-paint'].numericValue)).toBe(true);
+    });
+
+    it('should compute results with callback requestor', async () => {
+      const {page, serverBaseUrl} = state;
+      await page.goto(`${serverBaseUrl}/landing.html`);
+      const requestor = async () => {
+        page.click('a');
```

Comment:
which part of the following assertions lets us know the anchor was clicked? I see we check something about the byte weight of index.html down below, which works, but a more clear check would be to also look at requestedUrl/finalUrl

Reply:
also, page.click should be awaited.

That's a subtle bug I can see users running into. Anything we could do to prevent it? I don't think so... only case I can think of is _at least_ disallowing this function returning void (force an `async` or returning a promise) to nudge users, but that wouldn't have caught this example.

Reply:
> which part of the following assertions lets us know the anchor was clicked?

As far as a specific assertion, I don't think there is one without digging though the trace. IMO, the entire test is ensures the link is clicked because Lighthouse would time out/fail if a navigation is not triggered. `requestedUrl`/`finalUrl` checks sound good.

> That's a subtle bug I can see users running into. Anything we could do to prevent it?

No but I don't think it's a problem. Lighthouse waits for TTI after `requestor` is run. Even if `page.click` is not awaited, Lighthouse still triggers the navigation and waits for TTI.